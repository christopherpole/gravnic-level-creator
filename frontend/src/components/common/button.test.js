import React from 'react';
import { configure, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';

import Button from './button';

configure({ adapter: new Adapter() });

describe('The generic button component', () => {
  let props;

  beforeEach(() => {
    props = {};
  });

  it('Renders without exploding', () => {
    const button = shallow(<Button {...props} />);

    expect(button).toHaveLength(1);
  });

  it('Matches the current snapshot', () => {
    const button = shallow(<Button {...props} />);

    expect(toJson(button)).toMatchSnapshot();
  });

  it('Matches the current snapshot for a disabled button', () => {
    const button = shallow(<Button {...props} disabled />);

    expect(toJson(button)).toMatchSnapshot();
  });
});
