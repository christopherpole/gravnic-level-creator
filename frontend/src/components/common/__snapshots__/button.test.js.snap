// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> Matches the current snapshot 1`] = `
<styled.button
  disabled={false}
/>
`;

exports[`<Button /> Matches the current snapshot for a disabled button 1`] = `
<styled.button
  disabled={true}
/>
`;
